engine/flink/model-util/src/main/scala/pl/touk/nussknacker/engine/flink/util/transformer/aggregate/transformers.scala

```diff
@@ -86,10 +88,70 @@ object transformers {
             } else {
               keyedStream
                 .window(TumblingEventTimeWindows.of(Time.milliseconds(windowLength.toMillis)))
-                .aggregate(new UnwrappingAggregateFunction(aggregator))
+                .aggregate(new UnwrappingAggregateFunction(aggregator, _.value))
             }
           ExplicitUidInOperatorsSupport.setUidIfNeed(explicitUidInStatefulOperators(ctx), ctx.nodeId)(statefulStream)
             .map(_.map(aggregator.alignToExpectedType(_, expectedType)))
         }))
 
+  def sessionWindowTransformer(keyBy: LazyParameter[CharSequence],
+                            aggregateBy: LazyParameter[AnyRef],
+                            aggregator: Aggregator,
+                            windowLength: Duration,
+                            endSessionCondition: LazyParameter[java.lang.Boolean],
+                            variableName: String,
+                            explicitUidInStatefulOperators: FlinkCustomNodeContext => Boolean
+                           )(implicit nodeId: NodeId): ContextTransformation =
+      ContextTransformation.definedBy(aggregator.toContextTransformation(variableName, aggregateBy))
+        .implementedBy(
+          FlinkCustomStreamTransformation((start: DataStream[NkContext], ctx: FlinkCustomNodeContext) => {
+            val storedAggregateType = aggregator.computeStoredType(aggregateBy.returnType).valueOr(msg => throw new IllegalArgumentException(msg))
+            val expectedType = aggregator.computeOutputType(aggregateBy.returnType).valueOr(msg => throw new IllegalArgumentException(msg))
+            val keyedStream = start
+              .map(new SessionMapper(ctx.lazyParameterHelper, keyBy, aggregateBy, endSessionCondition))
+              .keyBy(_.value.key)
+            val statefulStream =
+                keyedStream
+                  .window(EventTimeSessionWindows.withGap(Time.milliseconds(windowLength.toMillis)))
+                  .trigger(new EndEventOrTimeTrigger[ValueWithContext[KeyedValue[String, (AnyRef, java.lang.Boolean)]], TimeWindow](EventTimeTrigger.create(), _.value.value._2))
+                  .aggregate(new UnwrappingAggregateFunction(aggregator, _.value._1))
```

Comment:
I wonder if it would be a good idea to extract aggregate function to some method to make easier to add aggregation saving session start time